aspnetcore/blazor/javascript-interoperability/import-export-interop.md

```diff
@@ -0,0 +1,378 @@
+---
+title: .NET JavaScript `[JSImport]`/`[JSExport]` interop with ASP.NET Core Blazor WebAssembly
+author: guardrex
+description: Learn how to interact with JavaScript in Blazor WebAssembly apps using .NET JavaScript `[JSImport]`/`[JSExport]` interop.
+monikerRange: '= aspnetcore-7.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 09/28/2022
+uid: blazor/js-interop/import-export-interop
+---
+# .NET JavaScript `[JSImport]`/`[JSExport]` interop with ASP.NET Core Blazor
+
+This article explains how to interact with JavaScript (JS) in Blazor WebAssembly apps using .NET JS `[JSImport]`/`[JSExport]` interop API released with .NET 7.
+
+Blazor provides its own JS interop mechanism based on the <xref:Microsoft.JSInterop.IJSRuntime> interface, which is uniformly supported across Blazor hosting models and described in the following articles:
+
+* <xref:blazor/js-interop/call-javascript-from-dotnet>
+* <xref:blazor/js-interop/call-dotnet-from-javascript>
+
+<xref:Microsoft.JSInterop.IJSRuntime> enables library authors to build JS interop libraries that can be shared across the Blazor ecosystem and remains the recommend approach for JS interop in Blazor.
+
+This article describes an alternative JS interop approach available for the first time with the release of .NET 7. The approaches described in this article can be used in place of existing <xref:Microsoft.JSInterop.IJSRuntime> approaches and should be used to replace obsolete unmarshalled JS interop API when migrating to .NET 7.
+
+## Obsolete JavaScript interop API
+
+Unmarshalled JS interop using <xref:Microsoft.JSInterop.IJSUnmarshalledRuntime> API is obsolete in ASP.NET Core 7.0. Follow the guidance in this article to replace the obsolete API.
+
+## Prerequisites
+
+[!INCLUDE[](~/includes/7.0-SDK.md)]
+
+## Cache busting during development
+
+When implementing JS interop, browser caching of JS files during development can interfere with writing and testing JS interop code. To force a browser to reload JS files (cache busting) during development, we recommend using browser [developer tools](https://developer.mozilla.org/docs/Glossary/Developer_Tools) with static asset caching disabled. For more information, access the documentation for the developer tools associated with your browser:
+
+* [Chrome DevTools](https://developer.chrome.com/docs/devtools/)
+* [Firefox Developer Tools](https://developer.mozilla.org/docs/Tools)
+* [Microsoft Edge Developer Tools overview](/microsoft-edge/devtools-guide-chromium/)
+
+## Enable unsafe blocks
```

Comment:
@pavelsavara @lewing @SteveSandersonMS  Is there a way we can restrict this to just the code generator so it doesn't impact the rest of the user's app?

Reply:
No AFAIK

Reply:
They could separate the interop code to separate assembly.

Reply:
But that's probably overkill.

Reply:
Let's discuss offline if there's something we could do to improve this. For example could we add a way to enable unsafe blocks for only the generated code? But we don't need to block this PR on this discussion.